### PR TITLE
[Feature-10273][Datasource] Add dataSource for AWS Athena

### LINKED ISSUE #10273: [Feature][Datasource] Add dataSource for AWS Athena
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Add dataSource for AWS Athena

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by guodongym**
## Purpose of the pull request

add dataSource for AWS Athena    

close  #10273 

## Brief change log

* Add the ATHENA backend
* Add the ATHENA test case
* Add the ATHENA frontend

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

### CODE REVIEW

**Review by EricGao888 — CHANGES_REQUESTED** [review]
Hi @guodongym , Thx for contributing this great feature! It seems there is no related docs for this PR, could you please add the docs? Thx again.

**Comment by guodongym** [comment]
> Hi @guodongym , Thx for contributing this great feature! It seems there is no related docs for this PR, could you please add the docs? Thx again.

Yes I can

**Comment by guodongym** [comment]
> Hi @guodongym , Thx for contributing this great feature! It seems there is no related docs for this PR, could you please add the docs? Thx again.

The document has been added

File: docs/docs/en/guide/datasource/athena.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# AWS Athena
+
+![mysql](../../../../img/new_ui/dev/datasource/athena.png)
```
Comment (zhongjiajie) [inline docs/docs/en/guide/datasource/athena.md]:
```suggestion
![athena](../../../../img/new_ui/dev/datasource/athena.png)
```

File: dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-athena/src/main/java/org/apache/dolphinscheduler/plugin/datasource/athena/param/AthenaDataSourceProcessor.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    public BaseDataSourceParamDTO createDatasourceParamDTO(String connectionJson) {
+        AthenaConnectionParam
+            connectionParams = (AthenaConnectionParam) this.createConnectionParams(connectionJson);
+
+        AthenaDataSourceParamDTO
+            athenaDatasourceParamDTO = new AthenaDataSourceParamDTO();
+        athenaDatasourceParamDTO.setAwsRegion(connectionParams.getAwsRegion());
+        athenaDatasourceParamDTO.setDatabase(connectionParams.getDatabase());
+        athenaDatasourceParamDTO.setUserName(connectionParams.getUser());
+        athenaDatasourceParamDTO.setOther(this.parseOther(connectionParams.getOther()));
```
Comment (zhongjiajie) [inline dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-athena/src/main/java/org/apache/dolphinscheduler/plugin/datasource/athena/param/AthenaDataSourceProcessor.java:59]: Can we add a new constructor for the class AthenaDataSourceParamDTO? we pass AthenaConnectionParam to it and   it see attribute inside the constructor?
I find out we all data source using the `setAttr` to set attribute, maybe we should discuss it

File: dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-athena/src/main/java/org/apache/dolphinscheduler/plugin/datasource/athena/param/AthenaDataSourceProcessor.java [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+    @Override
+    protected void checkHost(String host) {
+        // Do not need to set the host, nothing to do
+    }
+
+    @Override
+    protected void checkDatabasePatter(String database) {
+        // Do not need to set the database, nothing to do
+    }
```
Comment (zhongjiajie) [inline dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-athena/src/main/java/org/apache/dolphinscheduler/plugin/datasource/athena/param/AthenaDataSourceProcessor.java:157]: we can remove those method

File: docs/docs/en/guide/datasource/athena.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+| Description | Enter a description of the DataSource. |
+| Username | Set the AWS access key. |
+| Password | Set the AWS secret access key. |
+| AwsRegion | Set the AWS region. |
+| Database name | Enter the database name of the ATHENA connection. |
+| Jdbc connection parameters | Parameter settings for ATHENA connection, in JSON format. |
+
+## Native Supported
+
+No, read section example in [datasource-setting](../howto/datasource-setting.md) `DataSource Center` section to activate this datasource.
```
Comment (zhongjiajie) [inline docs/docs/en/guide/datasource/athena.md]: If not native supported, can we provider the JDBC connection name to download, better add some download link for it

File: dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-athena/src/main/java/org/apache/dolphinscheduler/plugin/datasource/athena/param/AthenaDataSourceProcessor.java (hunk 3 above)
Comment (guodongym) [reply dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-athena/src/main/java/org/apache/dolphinscheduler/plugin/datasource/athena/param/AthenaDataSourceProcessor.java:157]: if remove those method, will cause the parent class parameter check error, `org.apache.dolphinscheduler.plugin.datasource.api.datasource.AbstractDataSourceProcessor#checkDatasourceParam` , I find  POSTGRESQL  are also problems with database verification

File: dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-athena/src/main/java/org/apache/dolphinscheduler/plugin/datasource/athena/param/AthenaDataSourceProcessor.java (hunk 2 above)
Comment (guodongym) [reply dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-athena/src/main/java/org/apache/dolphinscheduler/plugin/datasource/athena/param/AthenaDataSourceProcessor.java:59]: I find out we all data source using the `setAttr` to set attribute, so i implemented in the same way,I'm not sure if it needs to be changed uniformly

**Comment by caishunfeng** [comment]
> SonarCloud Quality Gate failed.    [Quality Gate failed](https://sonarcloud.io/dashboard?id=apache-dolphinscheduler&pullRequest=11157)
> 
> [Bug](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=11157&resolved=false&types=BUG) [C](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=11157&resolved=false&types=BUG) [21 Bugs](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=11157&resolved=false&types=BUG) [Vulnerability](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=11157&resolved=false&types=VULNERABILITY) [E](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=11157&resolved=false&types=VULNERABILITY) [9 Vulnerabilities](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=11157&resolved=false&types=VULNERABILITY) [Security Hotspot](https://sonarcloud.io/project/security_hotspots?id=apache-dolphinscheduler&pullRequest=11157&resolved=false&types=SECURITY_HOTSPOT) [E](https://sonarcloud.io/project/security_hotspots?id=apache-dolphinscheduler&pullRequest=11157&resolved=false&types=SECURITY_HOTSPOT) [37 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=apache-dolphinscheduler&pullRequest=11157&resolved=false&types=SECURITY_HOTSPOT) [Code Smell](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=11157&resolved=false&types=CODE_SMELL) [A](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=11157&resolved=false&types=CODE_SMELL) [2101 Code Smells](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=11157&resolved=false&types=CODE_SMELL)
> 
> [](https://sonarcloud.io/component_measures?id=apache-dolphinscheduler&pullRequest=11157&metric=new_coverage&view=list) [37.9% Coverage](https://sonarcloud.io/component_measures?id=apache-dolphinscheduler&pullRequest=11157&metric=new_coverage&view=list) [](https://sonarcloud.io/component_measures?id=apache-dolphinscheduler&pullRequest=11157&metric=new_duplicated_lines_density&view=list) [2.4% Duplication](https://sonarcloud.io/component_measures?id=apache-dolphinscheduler&pullRequest=11157&metric=new_duplicated_lines_density&view=list)

I found the sonar result is wrong, seems checked all the code, how can we retry this check again?

**Comment by guodongym** [comment]
> > SonarCloud Quality Gate failed.    [Quality Gate failed](https://sonarcloud.io/dashboard?id=apache-dolphinscheduler&pullRequest=11157)
> > [Bug](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=11157&resolved=false&types=BUG) [C](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=11157&resolved=false&types=BUG) [21 Bugs](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=11157&resolved=false&types=BUG) [Vulnerability](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=11157&resolved=false&types=VULNERABILITY) [E](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=11157&resolved=false&types=VULNERABILITY) [9 Vulnerabilities](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=11157&resolved=false&types=VULNERABILITY) [Security Hotspot](https://sonarcloud.io/project/security_hotspots?id=apache-dolphinscheduler&pullRequest=11157&resolved=false&types=SECURITY_HOTSPOT) [E](https://sonarcloud.io/project/security_hotspots?id=apache-dolphinscheduler&pullRequest=11157&resolved=false&types=SECURITY_HOTSPOT) [37 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=apache-dolphinscheduler&pullRequest=11157&resolved=false&types=SECURITY_HOTSPOT) [Code Smell](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=11157&resolved=false&types=CODE_SMELL) [A](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=11157&resolved=false&types=CODE_SMELL) [2101 Code Smells](https://sonarcloud.io/project/issues?id=apache-dolphinscheduler&pullRequest=11157&resolved=false&types=CODE_SMELL)
> > [](https://sonarcloud.io/component_measures?id=apache-dolphinscheduler&pullRequest=11157&metric=new_coverage&view=list) [37.9% Coverage](https://sonarcloud.io/component_measures?id=apache-dolphinscheduler&pullRequest=11157&metric=new_coverage&view=list) [](https://sonarcloud.io/component_measures?id=apache-dolphinscheduler&pullRequest=11157&metric=new_duplicated_lines_density&view=list) [2.4% Duplication](https://sonarcloud.io/component_measures?id=apache-dolphinscheduler&pullRequest=11157&metric=new_duplicated_lines_density&view=list)
> 
> I found the sonar result is wrong, seems checked all the code, how can we retry this check again?

Done

**Review by caishunfeng — DISMISSED** [review]
LGTM of the backend part.

**Comment by zhongjiajie** [comment]
@songjianet @Amy0104 @devosend do you have time to take a look the FE part?

**Review by devosend — DISMISSED** [review]
LGTM for front end

**Comment by zhongjiajie** [comment]
I add missing doc side bar in 8def3b2

**Comment by zhongjiajie** [comment]
And will merge PR when it pass

**Comment by zhongjiajie** [comment]
Restart the failed test

File: dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-athena/src/main/java/org/apache/dolphinscheduler/plugin/datasource/athena/param/AthenaDataSourceParamDTO.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.dolphinscheduler.plugin.datasource.athena.param;
+
+import org.apache.dolphinscheduler.plugin.datasource.api.datasource.BaseDataSourceParamDTO;
+import org.apache.dolphinscheduler.spi.enums.DbType;
+
```
Comment (EricGao888) [inline dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-athena/src/main/java/org/apache/dolphinscheduler/plugin/datasource/athena/param/AthenaDataSourceParamDTO.java:22]: We could use `lombok` `@Data` here to help the data class look neat.

File: dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-athena/src/test/java/org/apache/dolphinscheduler/plugin/datasource/athena/param/AthenaDataSourceProcessorTest.java [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+@RunWith(PowerMockRunner.class)
```
Comment (EricGao888) [inline dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-athena/src/test/java/org/apache/dolphinscheduler/plugin/datasource/athena/param/AthenaDataSourceProcessorTest.java:39]: Better avoid using `powermock`, see: https://github.com/apache/dolphinscheduler/issues/11405 . But it's fine to me if you don't want to rewrite it at this moment. I could refactor this later in issue #11405

**Review by EricGao888 — DISMISSED** [review]
Backend part LGTM except few nits.

File: dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-athena/src/main/java/org/apache/dolphinscheduler/plugin/datasource/athena/param/AthenaDataSourceParamDTO.java (hunk 5 above)
Comment (guodongym) [reply dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-athena/src/main/java/org/apache/dolphinscheduler/plugin/datasource/athena/param/AthenaDataSourceParamDTO.java:22]: Done

File: dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-athena/src/test/java/org/apache/dolphinscheduler/plugin/datasource/athena/param/AthenaDataSourceProcessorTest.java (hunk 6 above)
Comment (guodongym) [reply dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-athena/src/test/java/org/apache/dolphinscheduler/plugin/datasource/athena/param/AthenaDataSourceProcessorTest.java:39]: waiting for you to refactor later

File: dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-athena/src/main/java/org/apache/dolphinscheduler/plugin/datasource/athena/param/AthenaDataSourceProcessor.java (hunk 2 above)
Comment (zhongjiajie) [reply dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-athena/src/main/java/org/apache/dolphinscheduler/plugin/datasource/athena/param/AthenaDataSourceProcessor.java:59]: get it

File: dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-athena/src/main/java/org/apache/dolphinscheduler/plugin/datasource/athena/param/AthenaDataSourceProcessor.java (hunk 3 above)
Comment (zhongjiajie) [reply dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-athena/src/main/java/org/apache/dolphinscheduler/plugin/datasource/athena/param/AthenaDataSourceProcessor.java:157]: 👌

File: dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-athena/src/main/java/org/apache/dolphinscheduler/plugin/datasource/athena/param/AthenaDataSourceParamDTO.java (hunk 5 above)
Comment (zhongjiajie) [reply dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-athena/src/main/java/org/apache/dolphinscheduler/plugin/datasource/athena/param/AthenaDataSourceParamDTO.java:22]: resolve

File: dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-athena/src/test/java/org/apache/dolphinscheduler/plugin/datasource/athena/param/AthenaDataSourceProcessorTest.java (hunk 6 above)
Comment (zhongjiajie) [reply dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-athena/src/test/java/org/apache/dolphinscheduler/plugin/datasource/athena/param/AthenaDataSourceProcessorTest.java:39]: mark resolve

**Review by zhongjiajie — APPROVED** [review]
LGTM, do you have further suggestions? @EricGao888

**Pull request merged**